gapis/api/gles/api/draw_commands.api

```diff
@@ -389,8 +414,13 @@ sub void DrawCommandDependencies(ref!Context ctx) {
     ReadFramebufferAttachment(read_fb.DepthAttachment)
   }
 
-  // The program execution read uniforms
   for _, _, u in ctx.Bound.Program.UniformLocations {
+    texture_target := GetTextureTargetFromSamplerType(u.Type)
+    if texture_target != GL_NONE {
+      DependencyReadTexture(ctx, GetBoundTextureForUnit(ctx.Bound.TextureUnit, texture_target))
```

Comment:
This is not correct. `ctx.Bound.TextureUnit` is the last "activated" texture unit using `glActiveTexture`. However, the shader will read from the texture unit referenced by the uniform's value. E.g. if the uniform has value '3' it will read from `ctx.TextureUnits[3]`.

This is going to get complicated with uniform buffers... Maybe will have to pretend as though the shader will read from every active texture unit... sigh. For now, though, use the uniform's value and we'll deal with uniform buffers later.

Reply:
Many thanks for this clarification, done!